tiledb/sm/query/ast/query_ast.h

```diff
@@ -205,13 +215,66 @@ class ASTNodeVal : public ASTNode {
     }
   };
 
+  /**
+   * @brief Copy constructor.
+   */
+  ASTNodeVal(const ASTNodeVal& rhs)
+      : field_name_(rhs.field_name_)
+      , condition_value_data_(rhs.condition_value_data_)
+      , condition_value_view_(
+            (rhs.condition_value_view_.content() != nullptr &&
+                     rhs.condition_value_view_.size() == 0 ?
+                 (void*)"" :
+                 condition_value_data_.data()),
+            condition_value_data_.size())
+      , op_(rhs.op_) {
+  }
+
+  ASTNodeVal(const ASTNodeVal& rhs, ASTNegationT)
+      : field_name_(rhs.field_name_)
+      , condition_value_data_(rhs.condition_value_data_)
+      , condition_value_view_(
+            (rhs.condition_value_view_.content() != nullptr &&
```

Comment:
This looks equivalent to `condition_value_view_(condition_value_data_.data(), condition_value_data_.size())`, or am I missing something here?

Reply:
(and same below)

Reply:
In the case of empty string (not null), condition_value_data_ will be unallocated, so we don't want the datum view to point to nullptr but "".